python_modules/libraries/dagster-tableau/dagster_tableau/assets.py

```diff
@@ -0,0 +1,93 @@
+from typing import Optional, Sequence, cast
+
+from dagster import (
+    AssetExecutionContext,
+    AssetsDefinition,
+    AssetSpec,
+    ObserveResult,
+    Output,
+    multi_asset,
+)
+
+from dagster_tableau.resources import BaseTableauWorkspace
+
+
+def build_tableau_executable_assets_definition(
+    resource_key: str,
+    specs: Sequence[AssetSpec],
+    refreshable_workbook_ids: Optional[Sequence[str]] = None,
+) -> AssetsDefinition:
+    """Returns the AssetsDefinition of the executable assets in the Tableau workspace.
+
+    Args:
+        resource_key (str): The resource key to use for the Tableau resource.
+        specs (Sequence[AssetSpec]): The asset specs of the executable assets in the Tableau workspace.
+        refreshable_workbook_ids (Optional[Sequence[str]]): A list of workbook IDs. The workbooks provided must
+            have extracts as data sources and be refreshable in Tableau.
+
+            When materializing your Tableau assets, the workbooks provided are refreshed,
+            refreshing their sheets and dashboards before pulling their data in Dagster.
+
+            This feature is equivalent to selecting Refreshing Extracts for a workbook in Tableau UI
+            and only works for workbooks for which the data sources are extracts.
+            See https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref_workbooks_and_views.htm#update_workbook_now
+            for documentation.
+
+    Returns:
+        AssetsDefinition: The AssetsDefinition of the executable assets in the Tableau workspace.
+    """
+
+    @multi_asset(
+        name=f"tableau_sync_site_{resource_key}",
+        compute_kind="tableau",
+        can_subset=False,
+        specs=specs,
+        required_resource_keys={resource_key},
+    )
+    def asset_fn(context: AssetExecutionContext):
+        tableau = cast(BaseTableauWorkspace, getattr(context.resources, resource_key))
+        with tableau.get_client() as client:
+            refreshed_workbooks = set()
+            for refreshable_workbook_id in refreshable_workbook_ids or []:
```

Comment:
would still be nice to get this in the resource IMO, but can always do that later

Reply:
I agree, I implemented it but for some reasons tests are failing with the changes - looks like a problem related with `mock.patch`. Will merge this as-is for now but I will work on fixing this issue.